http/handler.go

```diff
@@ -567,18 +567,41 @@ func respondStandby(core *vault.Core, w http.ResponseWriter, reqURL *url.URL) {
 	w.WriteHeader(307)
 }
 
+// getTokenFromReq parse headers of the incoming request to extract token if present
+// it accepts Authorization Bearer (RFC6750) and X-Vault-Token header
+func getTokenFromReq(r *http.Request) (string, error) {
+	if token := r.Header.Get(consts.AuthHeaderName); token != "" {
+		return token, nil
+	}
+	if v := r.Header.Get("Authorization"); v != "" {
+		// Reference for Authorization header format: https://tools.ietf.org/html/rfc7236#section-3
+
+		if !strings.HasPrefix(v, "Bearer") {
+			return "", fmt.Errorf("the Authorization header scheme should be Bearer")
+		}
+		vals := strings.Split(strings.TrimSpace(strings.TrimPrefix(v, "Bearer")), " ")
```

Comment:
If we checked for the full "Bearer " prefix above with a space, could this just become `return v[7:], nil`? https://play.golang.org/p/Nttq9bx15lu

Reply:
One might send space separated value (not that it would make sense tho)
Still have to split(s, ' ')[0]
but if the aim is to optimize, we can probably check for the presence of a space in the string without allocation, return an error if present, return value if not

https://play.golang.org/p/NgRoBv-8Y1N
This address the comment of Calvin too I believe

Reply:
I updated the PR with a new version, with less allocations